### PR TITLE
bindings and documentation

### PR DESCRIPTION
**Pull request by dkebler**
Joel,

In this branch dgk-bindings

1. Another take on the key and mouse bindings. 
  
    -   They make sense ergonomically  and metaphorically.  Key binding drills "down" into dependencies and "up" into dependents.  The support keys chosen also don't conflict with other linux and windows key bindings (ctrl alt arrows are used for windowing) and minimize finger/hand motion.
    -   I modified the OSX ones to match closely.  I can't get my OSX 10.6.8 machine to run dependents so you'll have to try them out.  See the new documentation :-).  I added a mouse binding for dependents.
    -   Before you have too many users now is the time to change these to minimize irritation.

2.  I rewrote a good portion of the documentation readme.md file, clarifying a few things and including the updated binding info and stuff on customizing, etc.  Hope you like it.

3.  I put a skeleton scss directory/file structure into the tests directory.  Automated tests are good but being old school having a simple manual testbed is nice.  There is a readme.md file in there with more info.

Be happy to make any changes you want and push up again before you merge.

David

### CODE REVIEW

File: Default (OSX).sublime-keymap [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
   // Jump to a dependency
   {
-    "keys": ["super+alt+right"],
+    "keys": ["super+alt+down"],
```
Comment (mrjoelkemp) [inline Default (OSX).sublime-keymap]: Unfortunately, that's already bound to `Go To Definition`.

File: Default (OSX).sublime-mousemap [hunk 2]
```diff
@@ -5,5 +5,12 @@
     "modifiers": ["super", "alt"],
     "press_command": "drag_select",
     "command": "jump_to_dependency"
+  },
+  {
+    "button": "button1",
+    "count": 1,
+    "modifiers": ["super", "shift"],
```
Comment (mrjoelkemp) [inline Default (OSX).sublime-mousemap]: +1

File: readme.md [hunk 3]
```diff
@@ -31,32 +33,76 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
 
 If it doesn't work, please file an issue.
 
+If you want contribute or use the latest commits simply git clone the repository `https://github.com/mrjoelkemp/Dependents` to your Packages directory with a directory name of `Dependents`. From sublime choose `Preferences -> Browse Packages`  to easily determine the path to the Packages directory for your install.  Be sure to remove the package via package control first if you had already done so to avoid package control updating the package.
+
+### Configuring settings
+
+There are two ways in which to invoke the package settings.  
+
+One is globally via entires in a user settings file `Preferences -> Package Settings -> Dependents -> Settings-User`.  You can open the `Settings-Default` file and copy and paste the contents as a starter template.   Do not edit the default file as that will be overwritten with updates.
```
Comment (mrjoelkemp) [inline readme.md]: I wanted to phase out the global settings from the readme because unless a dev uses the same structure for every project, this locks the editor into the configuration of a single project. The `.deprc` solution is significantly more flexible.

File: readme.md [hunk 4]
```diff
@@ -31,32 +33,76 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
...
+
+Alternatively you can define settings on a per-project basis by creating
+a `.deprc` file in the root of the sublime project/folder.  The paths you will specify are then **relative** to the sublime project/folder root. This is the preferred way if you are working on more than one project using sublime (aren't we all!)
+
+* `root`: (required) the ultimate/root path at which to limit dependent searching.  When using `.deprc` would be `""` if same as root of sublime project/folder.
```
Comment (mrjoelkemp) [inline readme.md]: Coming from the node.js world, I've always put `./`. I saw your issue opened for the empty root. Will have to look further, but we might want to consider suggesting `./` instead of the empty string.

Comment (mrjoelkemp) [inline readme.md]: I also believe the `root` is now optional as well. I made it that way for more sass-centric devs to use dependents without worrying about JS configuration.

File: readme.md [hunk 5]
```diff
@@ -31,32 +33,76 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
...
+ * Note: Subdirectories are not supported; you can't supply `some/sub/folder` as an exclusion
+* `build_config`: (Optional) path to your RequireJS Build configuration json file
+ * This should have a "modules" section that lists your bundles (entry points)
+ * Supplying this yields a significant performance speedup when finding relevant app entry points
+* `node_path`: (Optional) path to your node installation. Defaults to `/usr/local/bin` on OSX
+
+example:
+```
+{
+  "root": "public/assets/js",
```
Comment (mrjoelkemp) [inline readme.md]: I'll verify that `root` is indeed optional and if so, then the `#optional` suffix should go away and a bullet point could be made to stress that all options are optional – use what you need.

File: readme.md [hunk 6]
```diff
@@ -31,32 +33,76 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
...
+* Via mouse click [see the mouse bindings](#mouse-bindings)
+
+#### Jump to a dependency
+
+Quickly jump to a file that the current file @imports (sass) or requires (js)
+
+1. Within a file, place your cursor over the dependency path you want to go to
+2. Then trigger the `Jump to dependency` command in one of the four ways noted above.
+ 
+*For javascript* if a dependency is aliased, you'll need to supply the path to your requirejs configuration
```
Comment (mrjoelkemp) [inline readme.md]: +1

File: readme.md [hunk 7]
```diff
@@ -31,32 +33,76 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
...
 ### Usage
 
+There are four ways to trigger the package's commands
+
+* From Main Menu `File -> Dependents`
+* From Context `Right click -> Dependents`
+* Via keyboard [see the key bindings](#key-bindings)
+* Via mouse click [see the mouse bindings](#mouse-bindings)
+
+#### Jump to a dependency
```
Comment (mrjoelkemp) [inline readme.md]: Moved it up because that's your favorite feature? :)

I built the Dependents feature first, hence the name of the repo and the ordering (not tied to my original ordering at all – I like yours). In this order, it reminds me of music albums where the album name has a self-titled track somewhere amongst the track list :)

File: readme.md [hunk 8]
```diff
@@ -77,82 +123,178 @@ View a snapshot of the current file's dependency tree (as a JSON file)
...
+
+#### Mouse bindings
 
 By default, the following key bindings have been supplied:
 
 OSX:
 
-* Find Dependents: `CMD + Option + Up arrow`
-* Jump to dependency: `CMD + Option + Right arrow` or `CMD + Option + Click`
+* Jump to dependency: `Option + Command + Click` on the dependency item
```
Comment (mrjoelkemp) [inline readme.md]: I think the `Command` key being listed first is the norm for many OSX apps and ST plugins.

File: Default (Windows).sublime-keymap [hunk 9]
```diff
@@ -1,13 +1,13 @@
 [
   // Find the dependents for the current module
   {
-    "keys": ["control+alt+up"],
+    "keys": ["ctrl+shift+up"],
```
Comment (mrjoelkemp) [inline Default (Windows).sublime-keymap]: I really wanted uniformity about the key bindings, but as an OSX user, I'll defer to your judgement on more ergonomic key bindings.

File: tests/.deprc [hunk 10]
```diff
@@ -0,0 +1,4 @@
+  {
+    "root": "tests",
```
Comment (mrjoelkemp) [inline tests/.deprc]: Assuming you could also just do `./` if you're opening this up in its own directory

File: tests/scss/base/_base-theme.scss [hunk 11]
```diff
@@ -0,0 +1,11 @@
+/*  Base Theme  */
```
Comment (mrjoelkemp) [inline tests/scss/base/_base-theme.scss]: This is all pretty great. I wanted this for a bit but opened external projects for the longest time. I could even potentially write some automated tests using these fixtures. 

Many thanks!

**Comment by mrjoelkemp** [comment]
Great stuff here @dkebler. I'll make the edits and merge in soon.

Will ping you when merged and a subsequent release is cut.

Thanks for contributing.

File: Default (OSX).sublime-keymap (hunk 1 above)
Comment (dkebler) [inline Default (OSX).sublime-keymap]: cntrl-alt-right is a workspace command in linux.... yes all the different
os specific bindings are a pain and since they are outside of sublime a
pain to overrride.

On Wed, Mar 11, 2015 at 7:15 PM, Joel Kemp <notifications@github.com> wrote:

> In Default (OSX).sublime-keymap
> <https://github.com/mrjoelkemp/Dependents/pull/131#discussion_r26273833>:
>
> > @@ -7,7 +7,7 @@
> >
> >    // Jump to a dependency
> >    {
> > -    "keys": ["super+alt+right"],
> > +    "keys": ["super+alt+down"],
>
> Unfortunately, that's already bound to Go To Definition.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrjoelkemp/Dependents/pull/131/files#r26273833>.
>



-- 
Cheers, Cuidate, TTFN <http://www.youtube.com/watch?v=5Gu50vq5ux4>
David

File: tests/.deprc (hunk 10 above)
Comment (dkebler) [inline tests/.deprc]: yes this works

On Wed, Mar 11, 2015 at 7:46 PM, Joel Kemp <notifications@github.com> wrote:

> In tests/.deprc
> <https://github.com/mrjoelkemp/Dependents/pull/131#discussion_r26274763>:
>
> > @@ -0,0 +1,4 @@
> > +  {
> > +    "root": "tests",
>
> Assuming you could also just do ./ if you're opening this up in its own
> directory
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrjoelkemp/Dependents/pull/131/files#r26274763>.
>



-- 
Cheers, Cuidate, TTFN <http://www.youtube.com/watch?v=5Gu50vq5ux4>
David

File: Default (Windows).sublime-keymap (hunk 9 above)
Comment (dkebler) [inline Default (Windows).sublime-keymap]: ctrl is what is actually on a linux/windows keyboard and not the word
"control".
I wish we could be uniform!   Alas Steve Jobs had to "Think Different".
:-).
I also changed the order of osx commands more because of the actual key
locations (left to right) rather than some OSX semantics....likewise I must
defer to you on OSX.

On Wed, Mar 11, 2015 at 7:45 PM, Joel Kemp <notifications@github.com> wrote:

> In Default (Windows).sublime-keymap
> <https://github.com/mrjoelkemp/Dependents/pull/131#discussion_r26274702>:
>
> > @@ -1,13 +1,13 @@
> >  [
> >    // Find the dependents for the current module
> >    {
> > -    "keys": ["control+alt+up"],
> > +    "keys": ["ctrl+shift+up"],
>
> I really wanted uniformity about the key bindings, but as an OSX user,
> I'll defer to your judgement on more ergonomic key bindings.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrjoelkemp/Dependents/pull/131/files#r26274702>.
>



-- 
Cheers, Cuidate, TTFN <http://www.youtube.com/watch?v=5Gu50vq5ux4>
David

File: tests/scss/base/_base-theme.scss (hunk 11 above)
Comment (dkebler) [inline tests/scss/base/_base-theme.scss]: we should have something similar for each language supported.  Would be
helpful getting a start on js  to see an equivalent js structure.

On Wed, Mar 11, 2015 at 7:48 PM, Joel Kemp <notifications@github.com> wrote:

> In tests/scss/base/_base-theme.scss
> <https://github.com/mrjoelkemp/Dependents/pull/131#discussion_r26274807>:
>
> > @@ -0,0 +1,11 @@
> > +/*  Base Theme  */
>
> This is all pretty great. I wanted this for a bit but opened external
> projects for the longest time. I could even potentially write some
> automated tests using these fixtures.
>
> Many thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrjoelkemp/Dependents/pull/131/files#r26274807>.
>



-- 
Cheers, Cuidate, TTFN <http://www.youtube.com/watch?v=5Gu50vq5ux4>
David

File: readme.md (hunk 7 above)
Comment (dkebler) [inline readme.md]: yes I'm more inclined to have my root parent file open and drill down to
some code I want to edit.  But I see how dependents is "nicer" because in
an open file without good documentation you have no clue what are the
dependent(s).   So putting jump to first is more about the metaphor of
working down from a parent to it's offspring like in a genealogy.

On Wed, Mar 11, 2015 at 7:28 PM, Joel Kemp <notifications@github.com> wrote:

> In readme.md
> <https://github.com/mrjoelkemp/Dependents/pull/131#discussion_r26274247>:
>
> >  ### Usage
> >
> > +There are four ways to trigger the package's commands
> > +
> > +* From Main Menu `File -> Dependents`
> > +* From Context `Right click -> Dependents`
> > +* Via keyboard [see the key bindings](#key-bindings)
> > +* Via mouse click [see the mouse bindings](#mouse-bindings)
> > +
> > +#### Jump to a dependency
>
> Moved it up because that's your favorite feature? :)
>
> I built the Dependents feature first, hence the name of the repo and the
> ordering (not tied to my original ordering at all – I like yours). In this
> order, it reminds me of music albums where the album name has a self-titled
> track somewhere amongst the track list :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrjoelkemp/Dependents/pull/131/files#r26274247>.
>



-- 
Cheers, Cuidate, TTFN <http://www.youtube.com/watch?v=5Gu50vq5ux4>
David

**Comment by dkebler** [comment]
where can I pull your changes from??  (dgk-bindings at your github repo?)
I can go over them and update/fix things in all this that you sent me and
do an updated PR.  Happy to do this, save you some time.

On Wed, Mar 11, 2015 at 7:49 PM, Joel Kemp <notifications@github.com> wrote:

> Great stuff here @dkebler <https://github.com/dkebler>. I'll make the
> edits and merge in soon.
>
> Will ping you when merged and a subsequent release is cut.
>
> Thanks for contributing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrjoelkemp/Dependents/pull/131#issuecomment-78414792>.
>



-- 
Cheers, Cuidate, TTFN <http://www.youtube.com/watch?v=5Gu50vq5ux4>
David

File: readme.md (hunk 3 above)
Comment (dkebler) [inline readme.md]: From my perspective just remove it now.  It makes little sense and is of
little use offering the global option and only makes for more confusion.
Related would be to remove the menu item preferences->packages
settings->dependents so folks aren't tempted.

On Wed, Mar 11, 2015 at 7:21 PM, Joel Kemp <notifications@github.com> wrote:

> In readme.md
> <https://github.com/mrjoelkemp/Dependents/pull/131#discussion_r26274021>:
>
> > @@ -31,32 +33,76 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
> >
> >  If it doesn't work, please file an issue.
> >
> > +If you want contribute or use the latest commits simply git clone the repository `https://github.com/mrjoelkemp/Dependents` <https://github.com/mrjoelkemp/Dependents> to your Packages directory with a directory name of `Dependents`. From sublime choose `Preferences -> Browse Packages`  to easily determine the path to the Packages directory for your install.  Be sure to remove the package via package control first if you had already done so to avoid package control updating the package.
> > +
> > +### Configuring settings
> > +
> > +There are two ways in which to invoke the package settings.
> > +
> > +One is globally via entires in a user settings file `Preferences -> Package Settings -> Dependents -> Settings-User`.  You can open the `Settings-Default` file and copy and paste the contents as a starter template.   Do not edit the default file as that will be overwritten with updates.
>
> I wanted to phase out the global settings from the readme because unless a
> dev uses the same structure for every project, this locks the editor into
> the configuration of a single project. The .deprc solution is
> significantly more flexible.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrjoelkemp/Dependents/pull/131/files#r26274021>.
>



-- 
Cheers, Cuidate, TTFN <http://www.youtube.com/watch?v=5Gu50vq5ux4>
David

File: readme.md (hunk 5 above)
Comment (dkebler) [inline readme.md]: yes confirmed. root is option as ./ works

On Wed, Mar 11, 2015 at 7:25 PM, Joel Kemp <notifications@github.com> wrote:

> In readme.md
> <https://github.com/mrjoelkemp/Dependents/pull/131#discussion_r26274171>:
>
> > +* `root`: (required) the ultimate/root path at which to limit dependent searching.  When using `.deprc` would be `""` if same as root of sublime project/folder.
> > +* `config`: (Optional) the path to your requirejs configuration file
> > +* `sass_root`: (Optional) the path to your Sass files
> > +* `exclude`: (Optional) a list of paths and/or folder names to exclude from the search for dependents
> > + * Omitting folders that contain 3rd party libraries can drastically speed up the search for a large codebase.
> > + * The following folders are excluded by default: `node_modules`, `bower_components`, `vendor`
> > + * Note: Subdirectories are not supported; you can't supply `some/sub/folder` as an exclusion
> > +* `build_config`: (Optional) path to your RequireJS Build configuration json file
> > + * This should have a "modules" section that lists your bundles (entry points)
> > + * Supplying this yields a significant performance speedup when finding relevant app entry points
> > +* `node_path`: (Optional) path to your node installation. Defaults to `/usr/local/bin` on OSX
> > +
> > +example:
> > +```
> > +{
> > +  "root": "public/assets/js",
>
> I'll verify that root is indeed optional and if so, then the #optional
> suffix should go away and a bullet point could be made to stress that all
> options are optional – use what you need.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrjoelkemp/Dependents/pull/131/files#r26274171>.
>



-- 
Cheers, Cuidate, TTFN <http://www.youtube.com/watch?v=5Gu50vq5ux4>
David

File: readme.md (hunk 4 above)
Comment (dkebler) [inline readme.md]: confirmed.  root is optional and ./ allows  root of dependents and root
in sublime window to work properly.
So just need to document this properly and you can close that issue.

 {
    "sass_root": "./"
  }

On Wed, Mar 11, 2015 at 7:24 PM, Joel Kemp <notifications@github.com> wrote:

> In readme.md
> <https://github.com/mrjoelkemp/Dependents/pull/131#discussion_r26274125>:
>
> > @@ -31,32 +33,76 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
> >
> >  If it doesn't work, please file an issue.
> >
> > +If you want contribute or use the latest commits simply git clone the repository `https://github.com/mrjoelkemp/Dependents` <https://github.com/mrjoelkemp/Dependents> to your Packages directory with a directory name of `Dependents`. From sublime choose `Preferences -> Browse Packages`  to easily determine the path to the Packages directory for your install.  Be sure to remove the package via package control first if you had already done so to avoid package control updating the package.
> > +
> > +### Configuring settings
> > +
> > +There are two ways in which to invoke the package settings.
> > +
> > +One is globally via entires in a user settings file `Preferences -> Package Settings -> Dependents -> Settings-User`.  You can open the `Settings-Default` file and copy and paste the contents as a starter template.   Do not edit the default file as that will be overwritten with updates.
> > +
> > +Alternatively you can define settings on a per-project basis by creating
> > +a `.deprc` file in the root of the sublime project/folder.  The paths you will specify are then **relative** to the sublime project/folder root. This is the preferred way if you are working on more than one project using sublime (aren't we all!)
> > +
> > +* `root`: (required) the ultimate/root path at which to limit dependent searching.  When using `.deprc` would be `""` if same as root of sublime project/folder.
>
> I also believe the root is now optional as well. I made it that way for
> more sass-centric devs to use dependents without worrying about JS
> configuration.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrjoelkemp/Dependents/pull/131/files#r26274125>.
>



-- 
Cheers, Cuidate, TTFN <http://www.youtube.com/watch?v=5Gu50vq5ux4>
David

**Comment by dkebler** [comment]
looks like you just commented and made no changes yourself so I will just do edits on my branch and push up and let you know when they are made

**Comment by mrjoelkemp** [comment]
Yeah, haven't had time to dig into your replies. I just did a review and
planned to get to the edits when I had time. I haven't started yet.

Feel free to make a change if you agree with a comment, but I'd like to
review the disagreements (if any) before you make those changes.

File: Default (OSX).sublime-keymap (hunk 1 above)
Comment (dkebler) [inline Default (OSX).sublime-keymap]: put back to "right"

Comment (dkebler) [inline Default (OSX).sublime-keymap]: it's not the same now as linux but not likely too many will be jumping back and forth from osx to linux boxes and getting pissed, and I like "down" much better :-).   

A thought:  Is super alt left bound to anything in OSX.  You might want to change dependents binding to that so it's bilaterally pleasing :-).

File: readme.md (hunk 3 above)
Comment (dkebler) [inline readme.md]: removed mention of global settings, rewrote a bit of the .deprc description and removed settings menu entry from main.sublime-menu.

Still struggling with the clarity here.   Do you mean for "root" now to refer to js root only?   In which case I would rename "root" to "js_root".   Each language added could have it's own "xxxx_root" parameter would be my suggestion.  I know you started out with js so that is why you used just "root".   How ever this goes this needs to be clearer.  Also I might suggest having changing likewise have a preface js_ for js related settings.  

Once again you started with js but now there is possibility of multi-language support so this would avoid confusion.  As to sass was thinking of a sass_assets, sass_packages or sass_vendor path setting(s) which would solve the issue of when someone does want to look other places for third party sass imports that have no explicit dependency path in the @import like is the case when using bower.

Bottom line is there are still some clarity usage issues with .deprc. This was where I spent the most time trying to get dependents to work correctly.   Take a look at my latest commit first.  For now I just separated js from sass parameters for clarity and added more examples.

File: readme.md (hunk 5 above)
Comment (dkebler) [inline readme.md]: not sure yo mean "all" are optional.  Either one needs a "root" or "sass_root" at a minimum.  I rewrote to stress this.

File: readme.md (hunk 8 above)
Comment (dkebler) [inline readme.md]: put it back the way you had it but left it as is for linux/windows.

**Comment by dkebler** [comment]
Pushed up a new commit with changes per you comments.  Take another look

File: Default (Windows).sublime-keymap (hunk 9 above)
Comment (mrjoelkemp) [inline Default (Windows).sublime-keymap]: So long as `ctrl` works in the key definition, I'm fine with it.

File: Main.sublime-menu [hunk 12]
```diff
@@ -26,40 +26,5 @@
                 ]
             }
         ]
-    },
```
Comment (mrjoelkemp) [inline Main.sublime-menu]: Ah, this would force users to use `.deprc` files yes?

File: readme.md [hunk 13]
```diff
@@ -31,32 +33,101 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
...
+  "build_config": "assets/js/build.json", 
+  "node_path": "/my/node/install/folder", 
+  "sass_root": "assets/sass",  
+}
+```
+
+Tip: If you like to open a separate windows in sublime for subdirectories of your project just put a `.deprc` file there too with the reduced/correct relative paths.   
+
+
+**javascript specific settings**
```
Comment (mrjoelkemp) [inline readme.md]: I'd put this up above the "say you open..." user story. This way, it serves as a TL;DR for folks who want to try a quick setup. In my experience, no one even wants to read the changelog messages that show up when the package updates.

I still think the user story is valuable, for folks that want more of a walkthrough.

File: readme.md [hunk 14]
```diff
@@ -31,32 +33,101 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
...
 
+If you want contribute or use the latest commits simply git clone the repository `https://github.com/mrjoelkemp/Dependents` to your Packages directory with a directory name of `Dependents`. From sublime choose `Preferences -> Browse Packages`  to easily determine the path to the Packages directory for your install.  Be sure to remove the package via package control first if you had already done so to avoid package control updating the package.
+
+### Configuring settings
+
+Rather than using a typical sublime settings file you define settings on a *per-project basis* by creating a `.deprc` file in a project (open folder) root. The paths you will specify are then **relative** to the folder although absolute paths can be used.  Following are a few examples to help you get started.  Below are the current settings that the package recognizes.
+
+Say you open a folder/project in sublime to `/path/to/projects/someproject`.  Put a .deprc file in the project root  `/someproject`.   
+
+For javascript if you want dependents to search all the way to this root (`/someproject`) then put this in the `.deprc` file
```
Comment (mrjoelkemp) [inline readme.md]: JavaScript

File: readme.md [hunk 15]
```diff
@@ -31,32 +33,101 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
 
 If it doesn't work, please file an issue.
 
+If you want contribute or use the latest commits simply git clone the repository `https://github.com/mrjoelkemp/Dependents` to your Packages directory with a directory name of `Dependents`. From sublime choose `Preferences -> Browse Packages`  to easily determine the path to the Packages directory for your install.  Be sure to remove the package via package control first if you had already done so to avoid package control updating the package.
+
+### Configuring settings
+
+Rather than using a typical sublime settings file you define settings on a *per-project basis* by creating a `.deprc` file in a project (open folder) root. The paths you will specify are then **relative** to the folder although absolute paths can be used.  Following are a few examples to help you get started.  Below are the current settings that the package recognizes.
+
+Say you open a folder/project in sublime to `/path/to/projects/someproject`.  Put a .deprc file in the project root  `/someproject`.
```
Comment (mrjoelkemp) [inline readme.md]: `.deprc` in backticks

File: readme.md [hunk 16]
```diff
@@ -31,32 +33,101 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
...
+{
+  "root": "./",
+}
+```
+Use `"./"` to signify the root and NOT just a `""`.
+
+But if your javascript code was all contained within a subdirectory of `/someproject` you could limit your dependents search there.
+```
+{
+   "root": "root": "assets/js"
```
Comment (mrjoelkemp) [inline readme.md]: `"root": "assets/js"`. Duplicated `"root"`

File: readme.md (hunk 7 above)
Comment (mrjoelkemp) [inline readme.md]: :+1:

File: readme.md [hunk 17]
```diff
@@ -31,32 +33,101 @@ Don't see it? Try reinstalling Package Control. Altenatively, add the repository
...
+Quickly jump to a file that the current file @imports (sass) or requires (js)
+
+1. Within a file, place your cursor over the dependency path you want to go to
+2. Then trigger the `Jump to dependency` command in one of the four ways noted above.
+ 
+*For javascript* if a dependency is aliased, you'll need to supply the path to your requirejs configuration
+
 #### Find the dependents of the current module
 
 Dependents are files that immediately depend/require/import the current file.
```
Comment (mrjoelkemp) [inline readme.md]: This should be `depend on or require/import`

**Comment by mrjoelkemp** [comment]
This looks good @dkebler. A few minor nits, but this is ready to merge when you make the Windows-specific edits discussed in #113.

**Comment by dkebler** [comment]
ok give it another look over see what you think.   Happy to fix any typos and nits before merge.

**Comment by mrjoelkemp** [comment]
Thanks again for contributing @dkebler. Merged. I squashed the commits into one, so will have to reset your local master branch for future PRs.

**Comment by dkebler** [comment]
almost all good.

found a typo, has an extra "root":,  can you take care of that.

But if your JS code was all contained within a subdirectory of /someproject
you could limit your dependents search there.

{
   "root": "root": "assets/js"
}



I see the readme.md file is used at package control website, good.

Now that I think of it the documentation for windows install should note
that a git pull will be necessary to get the latest update since package
control is not used.   Might as well add that while fixing the typo.

On Mon, Mar 16, 2015 at 12:55 AM, Joel Kemp <notifications@github.com>
wrote:

> Thanks again for contributing @dkebler <https://github.com/dkebler>.
> Merged. I squashed the commits into one, so will have to reset your local
> master branch for future PRs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mrjoelkemp/Dependents/pull/131#issuecomment-81475126>.
>



-- 
Cheers, Cuidate, TTFN <http://www.youtube.com/watch?v=5Gu50vq5ux4>
David

**Comment by mrjoelkemp** [comment]
Will take care of it. thanks

**Comment by mrjoelkemp** [comment]
Fixed.
